pennylane/optimize/rotoselect.py

```diff
@@ -107,6 +107,7 @@ def step(self, objective_fn, x, generators):
             array: The new variable values :math:`x^{(t+1)}` as well as the new generators.
         """
         x_flat = np.fromiter(_flatten(x), dtype=float)
+        objective_fn_flat = lambda x_flat, gen: objective_fn(unflatten(x_flat, x), generators=gen)
```

Comment:
```suggestion
        # wrap the objective function so that it accepts the flattened parameter array
        objective_fn_flat = lambda x_flat, gen: objective_fn(unflatten(x_flat, x), generators=gen)
```